### PR TITLE
Add Friendships Model

### LINKED ISSUE #1: Setup databases
- [x] Events
- [x] Users
- [x] UserEvents
- [x] Friendships

### PR DESCRIPTION
**Pull request by lpile**
## Changes proposed in this pull request:
* Add Friendships model with spec and migration
* Add friendship relationships to User model with spec
* Update seeds file with friendships

## Card(s) closed in this pull request:
close #1 

## What did you struggle on to complete?
* Trying to set up a many to many which references one model.

### Test Coverage Percentage: 100%
- [x] Tested my new feature(s) as well as any feasible edge cases (if possible)
- [x] Checked coverage/index.html - did not add any new code that's not covered by testing (if possible)
- [x] Ran the test suite - all tests are passing
- [x] Merged in the latest master to my branch with git pull origin master & resolved merge conflicts
- [x] I have commented my code, particularly in hard-to-understand areas

## Helpful Resources:
* [Setting up Friendship Model](https://smartfunnycool.com/friendships-in-activerecord/)
* [Making Friendship rows unique at migration](https://stackoverflow.com/questions/19770888/rails-self-join-scheme-with-has-and-belongs-to-many)

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,9 +1,16 @@
 class User < ApplicationRecord
+  # Sets up relationship between user and event
   has_many :user_events
   has_many :events, through: :user_events
 
+  # Sets up relationship between user and other users called friends
+  has_many :friendships
+  has_many :inverse_friendships, :class_name => "Friendship", :foreign_key => "friend_id"
```
Comment (annekemcgrady) [inline app/models/user.rb]: Looks like you found the key here with this inverse_friendships?
